lib/puma/launcher.rb

```diff
@@ -539,5 +530,25 @@ def log_config
 
       log "\n"
     end
+
+    def original_bundler_env
+      return unless defined?(Bundler)
+
+      env = Bundler::ORIGINAL_ENV.dup
+      # add -rbundler/setup so we load from Gemfile when restarting
+      bundle = "-rbundler/setup"
+      env["RUBYOPT"] = [env["RUBYOPT"], bundle].join(" ").lstrip unless env["RUBYOPT"].to_s.include?(bundle)
+      env
+    end
+
+    def update_env(previous_env)
+      replacement_env = if @options[:reload_env_on_restart]
```

Comment:
Is this how the option would look like?  Unsure about the name.  Maybe `refresh_env_on_restart`?

Reply:
Either seems clear enough to me